library/src/test/java/com/datatorrent/lib/db/jdbc/JdbcInputOperatorApplicationTest.java

```diff
@@ -0,0 +1,179 @@
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import org.apache.hadoop.conf.Configuration;
+
+import com.datatorrent.api.Context;
+import com.datatorrent.api.DAG;
+import com.datatorrent.api.DefaultInputPort;
+import com.datatorrent.api.LocalMode;
+import com.datatorrent.api.StreamingApplication;
+import com.datatorrent.common.util.BaseOperator;
+import com.datatorrent.stram.StramLocalCluster;
+
+public class JdbcInputOperatorApplicationTest extends JdbcOperatorTest
+{
+  public static int TupleCount;
+
+  public int getNumOfRowsinTable(String tableName)
+  {
+    Connection con;
+    try {
+      con = DriverManager.getConnection(URL);
+      Statement stmt = con.createStatement();
+      String countQuery = "SELECT count(*) from " + tableName;
+      ResultSet resultSet = stmt.executeQuery(countQuery);
+      resultSet.next();
+      return resultSet.getInt(1);
+    } catch (SQLException e) {
+      throw new RuntimeException("fetching count", e);
+    }
+  }
+
+  @Test
+  public void testJdbcPOJOPollInputOperatorApplication() throws Exception
+  {
+    try {
+      LocalMode lma = LocalMode.newInstance();
+      Configuration conf = new Configuration(false);
+      conf.addResource(this.getClass().getResourceAsStream("/JdbcProperties.xml"));
+      lma.prepareDAG(new JdbcPOJOPollInputOperatorApplication(), conf);
+      LocalMode.Controller lc = lma.getController();
+      lc.setHeartbeatMonitoringEnabled(false);
+      ((StramLocalCluster)lc).setExitCondition(new Callable<Boolean>()
+      {
+        @Override
+        public Boolean call() throws Exception
+        {
+          return TupleCount == 10;
+        }
+      });
+
+      lc.run(10000);// runs for 10 seconds and quits
+      Assert.assertEquals("rows in db", TupleCount, getNumOfRowsinTable(TABLE_POJO_NAME));
+    } catch (ConstraintViolationException e) {
+      Assert.fail("constraint violations: " + e.getConstraintViolations());
+    }
+  }
+
+  @Test
+  public void testJdbcPOJOInputOperatorApplication() throws Exception
+  {
+    try {
+      LocalMode lma = LocalMode.newInstance();
+      Configuration conf = new Configuration(false);
+      conf.addResource(this.getClass().getResourceAsStream("/JdbcProperties.xml"));
+      lma.prepareDAG(new JdbcPOJOInputOperatorApplication(), conf);
+      LocalMode.Controller lc = lma.getController();
+      lc.setHeartbeatMonitoringEnabled(false);
+      ((StramLocalCluster)lc).setExitCondition(new Callable<Boolean>()
+      {
+        @Override
+        public Boolean call() throws Exception
+        {
+          return TupleCount == 10;
+        }
+      });
+
+      lc.run(10000);// runs for 10 seconds and quits
+      Assert.assertEquals("rows in db", TupleCount, getNumOfRowsinTable(TABLE_POJO_NAME));
+    } catch (ConstraintViolationException e) {
+      Assert.fail("constraint violations: " + e.getConstraintViolations());
+    }
+  }
+
+  public static class JdbcPOJOInputOperatorApplication implements StreamingApplication
```

Comment:
Lot of common code between these classes: ```JdbcPOJOInputOperatorApplication``` and ```JdbcPOJOPollInputOperatorApplication```. Please create methods to reduce code duplication.